detection/evasion/unexpected-kernel-extensions-macos.sql

```diff
@@ -17,3 +17,6 @@ WHERE
     AND name = '__kernel__'
   )
   AND exception_key NOT IN ('/Library/StagedExtensions/Library/Extensions/CalDigitUSBHubSupport.kext,com.CalDigit.USBHubSupport,1,<3>')
+  AND exception_key NOT LIKE '/Library/StagedExtensions/Library/Extensions/ufsd_NTFS.kext,com.paragon-software.filesystems.ntfs,%'
+  AND exception_key NOT LIKE '/Library/StagedExtensions/Library/Filesystems/macfuse.fs/Contents/Extensions/12/macfuse.kext,io.macfuse.filesystems.macfuse,%'
+  AND exception_key NOT LIKE '/Library/StagedExtensions/Library/Extensions/ufsd_ExtFS.kext,com.paragon-software.filesystems.extfs,%'
```

Comment:
Is there any concern with a malicious actor renaming their kext/extfs to match these rules?  I want to say that there's a way to list extensions by their developer ids instead?

Reply:
That's a great point - I don't know why we weren't already doing that here.

Feel free to send a PR if you are interested. My suggestion would is to change `exception_key` to be a combination of path, signature.authority, and signature.identifier.

You can see https://github.com/chainguard-dev/osquery-defense-kit/blob/main/detection/persistence/unexpected-listening-port-macos.sql as an example.

Reply:
This is one of those places where I'm not sure it's worth the effort  -- I /think/ Apple is only letting select kernel extensions through now?  Most everything is mean to be a system extension?

I'm also a bit fuzzy about "StagedExtensions" -- I thought this location was mean to be used when updating / installing extensions -- like they'd go here first, and then at next reboot, macos would move the stuff to /Library/Extensions -- I.E. the stuff in staged isn't actually used?